src/NServiceBus.Core/Recoverability/Faults/FaultToDispatchConnector.cs

```diff
@@ -25,7 +24,7 @@ public override Task Invoke(IFaultContext context, Func<IRoutingContext, Task> s
                 }
             }
 
-            var dispatchContext = new RoutingContext(message, new UnicastRoutingStrategy(context.ErrorQueueAddress), context);
+            var dispatchContext = this.CreateRoutingContext(context);
```

Comment:
And you could also use CreateRoutingContext(context, message, context.ErrorQueueAddress)